src/renderers/webgl/plugins/LensFlarePlugin.js

```diff
@@ -337,13 +339,15 @@ THREE.LensFlarePlugin = function ( renderer, flares ) {
 			screenPositionPixels.x = screenPosition.x * halfViewportWidth + halfViewportWidth;
 			screenPositionPixels.y = screenPosition.y * halfViewportHeight + halfViewportHeight;
 
+			areaToCopy.x = viewport.x + screenPositionPixels.x - 8; // horizontal coordinate of the lower left corner of the pixels to copy
+			areaToCopy.y = viewport.y + screenPositionPixels.y - 8; // vertical coordinate of the lower left corner of the pixels to copy
+
 			// screen cull
 
-			if ( hasVertexTexture || (
```

Comment:
If `hasVertexTexture` is true, the "logical OR" makes the evaluation of the viewport check irrelevant. I think this behavior is not correct. The viewport check should be the only criterion which determines rendering. `hasVertexTexture` is just a flag to load the correct vertex and fragment shader.